Sources/JTAppleCalendar/JTACInteractionMonthFunctions.swift

```diff
@@ -498,6 +498,74 @@ extension JTACMonthView {
             } else if yOffset >= contentSize.height - frame.height {
                 yOffset = contentSize.height - frame.height
             }
+            
+            // This was needed for when the user scrolls to next segment which will be the last,
+            // then the value is not exact the same as the height of the content,
+            // and then the didScrollToSegment delegate got called before the scroll happened, this fixed it.
+            
+            yOffset = yOffset.rounded()
+        }
+        
+        // Get valid indexPath of date to scroll to
+        
+        // This code is needed so that when we call the delegate we can give it the right DateSegmentInfo
+        var date: Date
+        let visibleDatesTMP = visibleDates()
+        
+        switch destination {
+            case .start:
+                date = startDateCache
+            case .end:
+                date = endDateCache
+            case .next:
+                let nextDate = visibleDatesTMP.outdates.last?.date ?? endDateCache
+                
+                // We add 5 days so that we are sure, that we are in the month
+                // (month end and start dates can sometimes result in wrong DateSegmentInfo)
+                let correctedNextDate = calendar.date(byAdding: .day, value: 5, to: nextDate) ?? endDateCache
+                
+                // If we added maybe too much days we check if we are in the range of the calendar
+                // if we gone too far we just use the endDate of the Calendar
+                if correctedNextDate < endDateCache
```

Comment:
You can change to: 

date = correctedNextDate < endDateCache ? correctedNextDate : endDateCache